internal/file/tar_test.go

```diff
@@ -0,0 +1,85 @@
+package file
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// looks like there isn't a helper for this yet? https://github.com/stretchr/testify/issues/497
```

Comment:
a missed opportunity from testify

Reply:
I followed a chain of linked issues/PRs, and it looks like maybe they have actually added this now? https://pkg.go.dev/github.com/stretchr/testify@v1.7.0/assert#ErrorAs

Reply:
nice! I just need to update testify.... good find 👍

Reply:
interesting, using it seems to be perplexing though:
```
=== RUN   TestSafeJoin_PositiveCases//a/place:[../_../]
    tar_test.go:86: 
        	Error Trace:	tar_test.go:86
        	Error:      	Should be in error chain:
        	            	expected: "paths are not allowed to resolve outside their prefix (\"/a/place\"). Destination: [\"../\" \"../\"]"
        	            	in chain: "paths are not allowed to resolve outside their prefix (\"/a/place\"). Destination: [\"../\" \"../\"]"
        	Test:       	TestSafeJoin_PositiveCases//a/place:[../_../]
```

shout out if you see the difference? (I don't see it)

In the meantime I'm going to leave the existing helper since it's generating an assertion which helps with the test table setup anyway.

Reply:
@wagoodman Yeah, I think you're right about the usefulness in the assertion helper function. In case it's useful, here's one implementation of that function that seems to work well:

```go
func assertErrorIsZipSlipDetected(t assert.TestingT, actualErr error, _ ...interface{}) bool {
	var expected *errZipSlipDetected
	return assert.ErrorAs(t, actualErr, &expected)
}
```

(and then down in the test case you'd see this...)

```go
{
// ...
	expected:     "",
	errAssertion: assertErrorIsZipSlipDetected,
},
```

Reply:
awesome! I think I'll leave the existing variant in place, but your reference implementation here shows what I did wrong (I passed the address for a value, not a pointer type). 🦖 🌮

Reply:
🦖 🌮